lib/index.js

```diff
@@ -47,8 +49,190 @@ api.validateConfiguration = (validatorConfig, callback) =>
+  // indicate that we actually want x number of one sig type, and y number of
+  // another sig type.
+  // For this first use case, we will not accept *only* 2 ld-signatures
+  // or *only* 2 equihash sigs.  `minimumProofsRequired` is currently being
+  // ignored
+
+  if(!hasOneOfEach) {
+    return callback(new BedrockError(
+      'The signature requirements have not been met.',
+      'ValidationError', {
+        httpStatusCode: 400,
+        public: true,
+        event,
+      }));
+  }
+
+  async.each(signature, (s, callback) => {
+    const sigZero = s[0];
+    const e = bedrock.util.clone(event);
+    e.signature = sigZero;
+    if(sigZero.type === 'LinkedDataSignature2015') {
+      return _verifyLinkedData(e, (err, result) => {
+        if(err) {
+          return callback(err);
+        }
+        if(!result.verified) {
+          callback(new BedrockError(
+            'LinkedData Signature(s) could not be verified.',
+            'ValidationError', {
+              httpStatusCode: 400,
+              public: true,
+              event,
+              keyResults: result.keyResults
+            }));
+        }
+        // success, do not return an error
+        callback();
+      });
+    }
+    if(sigZero.type === 'EquihashProof2017') {
+      return _verifyEquihash(e, {equihashParameterAlgorithm}, (err, result) => {
+        if(err) {
+          return callback(err);
+        }
+        if(!result) {
+          callback(new BedrockError(
+            'Equihash Signature could not be verified.',
+            'ValidationError', {
+              httpStatusCode: 400,
+              public: true,
+              event
+            }));
+        }
+        // success, do not return an error
+        callback();
+      });
+    }
+    // NOTE: this callback is for safety and *should* never be reached
+    callback(new BedrockError(
+      'Unsupported signature type.',
+      'OperationError', {
+        httpStatusCode: 400,
+        public: true,
+        event,
+      }));
+  }, err => {
+    console.log('EEEEEEEE', err);
+    callback(err);
+  });
+}
+
+function _verifyLinkedData(doc, callback) {
+  let cacheKey;
+  jsigs.verify(doc, {
+    getPublicKey: _getPublicKey,
+    getPublicKeyOwner: _getPublicKeyOwner
+  }, (err, result) => {
+    console.log('11111111', err);
+    console.log('22222222', result);
+    callback(err, result);
+  });
+
+  function _getPublicKey(keyId, options, callback) {
+
+    // FIXME: this seems very specific to the veres-one use case?
+
+    const publicKey = doc.input[0].authenticationCredential
+      .filter(k => k.id === keyId);
+    if(publicKey.length !== 1) {
+      return callback(new BedrockError(
+        'Public key not found.',
+        'NotFoundError', {
+          httpStatusCode: 404,
+          public: true,
+          keyId,
+        }));
+    }
+
+    // FIXME: should there already be a context on the key in the
+    // `authenticationCredential`?
```

Comment:
No, we should not presume that.

Reply:
Can you say more about this?  Are you saying it is normal that a context *must* be added in order to verify the signature?

Reply:
Solution: if no context present, copy from input[0], and add fixme to use framing in the future.

Reply:
Done.